File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] pre = new int[nums.length];
        int[] suf = new int[nums.length];
        int[] ans = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            if (i == 0) {
                pre[i] = nums[i];
            } else {
                pre[i] = nums[i] * pre[i - 1];
            }
        }
        for (int i = nums.length - 1; i >= 0; i--) {
            if (i == nums.length - 1) {
                suf[i] = nums[i];
            } else {
                suf[i] = nums[i] * suf[i + 1];
            }
        }
        for (int i = 0; i < nums.length; i++) {
            int p = 1;
            int s = 1;
            if (i - 1 >= 0) {
                p = pre[i - 1];
            }
            if (i + 1 < nums.length) {
                s = suf[i + 1];
            }
            ans[i] = p * s;
        }
        return ans;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        int freq[2005]={0};
        pair<int,int>pa[2005];
        vector<int>ans;
        for(int i=0;i<nums.size();i++)
        {
            freq[nums[i]+1000]++;
        }
        for(int i=0;i<2005;i++)
        {
            pa[i].first=freq[i];
            pa[i].second=i;
        }
        sort(pa ,pa +2005);
        for(int i=2004;i>=0&&k>0;i--)
        {
            k--;
            ans.push_back(pa[i].second-1000);
        }
        return ans;
    }
};
